### ISSUE TITLE
Option to automatically create a new branch based on the currently checked out branch

### ISSUE THREAD

**Issue by medallyon** [desktop/desktop#14480]
### The feature request

I would like an option, where when I create a new branch, the selected branch it's based on is not `main`, but the currently checked out branch (in my case `develop`). This would save a lot of time because more often than not, I forget to select the option to base my new branch off of the currently checked out branch and waste a lot of time waiting for Desktop to create a new branch based on `main`, which may be vastly different from my currently checked out branch, where Desktop first checks out the `main` branch, attempts to copy it to a new branch, and sometimes even messes up some uncommitted changes.

### Proposed solution

A simple switch in **Options > Advanced** that sets the default setting for which branch to base the new branch on. This may be similar to the **`If I have changes and I switch branches...`** setting.

### Additional context

![image](https://user-images.githubusercontent.com/17483199/166059687-69cb8194-b785-4bc1-9805-951cb4fe55e3.png)

**Comment by tidy-dev** [comment]
I know this is not what you are asking for, but I have run into the same thing you are, and maybe this workaround will help. When you create a branch from a current branch, you are just creating a branch from its latest commit. Thus, one way you can achieve this, that is more direct, is to right click on the latest commit in your history and select "Create branch from Commit"

**Comment by niik** [comment]
Thanks for the issue @medallyon. This has been previously discussed in https://github.com/desktop/desktop/issues/9845, and [this comment](https://github.com/desktop/desktop/issues/9845#issuecomment-632591239) outlines the reasoning behind this decision. 

At this time we don't have any plans to change this behavior. I'm going to close this issue out, but we'd still value feedback going forward about this decision and we do continue to evaluate issues even if they are closed. Thanks for using GitHub Desktop!

**Comment by medallyon** [comment]
Thanks @niik, I missed that issue. I still believe that it's nice to have options, and this is exactly what options are for, right? My main argument aligns with that of the one in #9845, where I've ended up in a situation where I created a new branch based on master, throwing off my common workflow, especially because I'm also mostly familiar with git as a cli.

It may well be that the majority of users intend to only base their new branch off main in order to do a single pull request. This makes sense for the open-source mindset. But for small or private projects where there may only be a single contributor (like me), I believe the norm is that you would always want to base your new branch on the currently checked out branch.

This decision is contextual, and I think a panel of mostly open-source maintainers declaring that this is how it should be, without data to back that declaration up, is a bit mis-guided. People love their specific code editor because it allows customisation, and I think it's the same with software like Desktop.

I actually had a look at getting started on a PR for this myself, even if just to have a forked client that contains this setting, but I cannot for the life of me find the file that contains this dialog, or the file that deals with the settings. If I could get some guidance with that, that would be super neat.
